### PR TITLE
Make shell scripts work on Windows

### PR DESCRIPTION
**Pull request by Artoria2e5**
The first commit changes `run_shell_command` to always launch the shell as if it is a program. Doing so removes the interoperability problem with Python's interpretation of `shell = True` on different platforms. After this commit, `augur align` fails correctly with a 127 (command not found) exit value on Windows.

I have also modified to print the stdout/err on error, since doing so makes debug a bit easier.

The second commit shell-quotes all user-specified filenames. This is to avoid strange errors and scary "command injection vulnerabilities" with weird-looking filenames.

### CODE REVIEW

File: augur/utils.py [hunk 1]
```diff
@@ -544,21 +546,30 @@ def run_shell_command(cmd, raise_errors = False, extra_env = None):
...
         # Use check_call() instead of run() since the latter was added only in Python 3.5.
-        subprocess.check_call(
-            "set -euo pipefail; " + cmd,
-            shell = True,
-            executable = "/bin/bash",
-            env = env)
+        out = subprocess.check_output(
+            shellexec + shargs,
+            shell = False,
+            stderr=subprocess.STDOUT)
```
Comment (tsibley) [inline augur/utils.py]: This change from `check_call` to `check_output` drops the environment modification `env = env`. That's an important part that should stay.

File: augur/utils.py [hunk 2]
```diff
@@ -544,21 +546,30 @@ def run_shell_command(cmd, raise_errors = False, extra_env = None):
...
 
     except subprocess.CalledProcessError as error:
         print_error(
-            "shell exited {rc} when running: {cmd}{extra}",
+            "{out}\nshell exited {rc} when running: {cmd}{extra}",
+            out = out,
```
Comment (tsibley) [inline augur/utils.py]: Based on the doc for `check_output`, it seems like `out = out` needs to be `error.output`? https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError.output

File: augur/utils.py [hunk 3]
```diff
@@ -571,11 +582,13 @@ def run_shell_command(cmd, raise_errors = False, extra_env = None):
     except FileNotFoundError as error:
         print_error(
             """
+            {out}
             Unable to run shell commands using {shell}!
 
             Augur requires {shell} to be installed.  Please open an issue on GitHub
             <https://github.com/nextstrain/augur/issues/new> if you need assistance.
             """,
+            out   = out,
```
Comment (tsibley) [inline augur/utils.py]: Again, I don't think `out` will be defined if an exception was raised here?

File: augur/utils.py (hunk 1 above)
Comment (Artoria2e5) [reply augur/utils.py]: Oops!

File: augur/utils.py (hunk 2 above)
Comment (Artoria2e5) [reply augur/utils.py]:
```suggestion
            out = error.output,
```

File: augur/utils.py [hunk 4]
```diff
@@ -571,11 +582,13 @@ def run_shell_command(cmd, raise_errors = False, extra_env = None):
     except FileNotFoundError as error:
         print_error(
             """
+            {out}
```
Comment (Artoria2e5) [inline augur/utils.py]:
```suggestion
```

File: augur/utils.py (hunk 3 above)
Comment (Artoria2e5) [reply augur/utils.py]:
```suggestion
```

File: augur/utils.py [hunk 5]
```diff
@@ -544,21 +546,30 @@ def run_shell_command(cmd, raise_errors = False, extra_env = None):
     overlayed onto the default subprocess environment.
     """
     env = os.environ.copy()
+    out = ''
```
Comment (Artoria2e5) [inline augur/utils.py]:
```suggestion
```

File: augur/utils.py [hunk 6]
```diff
@@ -544,21 +546,30 @@ def run_shell_command(cmd, raise_errors = False, extra_env = None):
...
+            # We try best effort on other systems. For now that means nt/java.
+            shellexec = ['env', 'bash']
+
         # Use check_call() instead of run() since the latter was added only in Python 3.5.
-        subprocess.check_call(
-            "set -euo pipefail; " + cmd,
-            shell = True,
-            executable = "/bin/bash",
-            env = env)
+        out = subprocess.check_output(
```
Comment (Artoria2e5) [inline augur/utils.py]:
```suggestion
        subprocess.check_output(
```

**Comment by tsibley** [comment]
Thanks @Artoria2e5, merged!

**Pull request merged**